### ISSUE TITLE
Variable override seems buggy

### ISSUE THREAD

**Issue by yiwen101** [MarkBind/markbind#2519]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Tell us about your environment

MacOs

### MarkBind version

Current master

### Describe the bug and the steps to reproduce it

in _markbind/variables.md: 
```
<variable name="filename">variables.md</variable>
```

in index.md:
```
{% set filename = "index.md" %}
filename is {{ filename }}
{% set x = "index.md" %}
x is {{ x }}
```

### Expected behavior

I expect to see: 
"file name is index.md
  x is index.md",
  
but I actually see
"file name is variable.md
   x is index.md"

**Comment by damithc** [comment]
@yiwen101 To facilitate reuse, an outsider should be able to override variables to customize the component to fit her needs (yes, this is the opposite of normal variable overriding). That way, the component can be reused without changing its code.

**Comment by yiwen101** [comment]
> @yiwen101 To facilitate reuse, an outsider should be able to override variables to customize the component to fit her needs (yes, this is the opposite of normal variable overriding). That way, the component can be reused without changing its code.

I see, thanks for the clarification. 
I just noticed that the documentation also specifies that "Global variables (_markbind/variables.md) will take precedence over any variables set via Nunjucks' tags (e.g. {% set %}).". This is quite counter intuitive.

I found another [issue](https://github.com/MarkBind/markbind/issues/1261) related to this. So sorry for posting this is duplicate issue.
